src/gululog_idx.erl

```diff
@@ -234,6 +235,34 @@ delete_from_cache(Tid, LogId) ->
   end,
   Tid.
 
+%% @doc Truncate after given logid from cache and index file
+%% Return new index and index file for deleted or truncated
+%% @End
+-spec truncate(dirname(), index(), segid(), logid(), ?undef | dirname()) ->
+  {index(), [filename()]}.
+truncate(Dir, #idx{tid = Tid} = Idx, SegId, LogId, BackupDir) ->
+  Ms = ets:fun2ms(fun(?ETS_ENTRY(_, LogIdX, _) = EtsEntry) when LogIdX > LogId -> EtsEntry end),
+  case EtsEntryList = ets:select(Tid, Ms) of
```

Comment:
prefer this:
case ets:select(Tid, Ms) of
  [] ->
  EtsEntryList -> %% bind the variable while matching.
end.